recipes/eleuther_eval.py

```diff
@@ -163,101 +420,84 @@ class EleutherEvalRecipe(EvalRecipeInterface):
+
+        # Load checkpoint
+        checkpointer = config.instantiate(cfg.checkpointer)
+        if quantization_mode is None:
             ckpt_dict = checkpointer.load_checkpoint()
         else:
             # weights_only needs to be False when loading a quantized model
             # currently loading a quantized model is only supported with the
             # FullModelTorchTuneCheckpointer
             ckpt_dict = checkpointer.load_checkpoint(weights_only=False)
 
-        self._model = self._setup_model(
-            model_cfg=self._cfg.model,
-            model_state_dict=ckpt_dict[training.MODEL_KEY],
-        )
-        self._tokenizer = config.instantiate(self._cfg.tokenizer)
-        logger.info("Tokenizer is initialized from file.")
+        # Initialize model
+        with training.set_default_dtype(self.dtype), self.device:
+            model = config.instantiate(cfg.model)
 
-    def _setup_model(
-        self,
-        model_cfg: DictConfig,
-        model_state_dict: Dict[str, Any],
-    ) -> nn.Module:
-        with training.set_default_dtype(self._dtype), self._device:
-            model = config.instantiate(model_cfg)
-
-        if self._quantization_mode is not None:
-            model = self._quantizer.quantize(model)
-            model = model.to(device=self._device, dtype=self._dtype)
-            for k, v in model_state_dict.items():
-                model_state_dict[k] = v.to(self._device)
-            model.load_state_dict(model_state_dict, assign=True)
-        else:
-            model.load_state_dict(model_state_dict)
+        # Quantize model if requested
+        if quantization_mode is not None:
+            model = quantizer.quantize(model)
+            model = model.to(device=self.device, dtype=self.dtype)
+
+        # Load model weights into initialized model
+        model.load_state_dict(ckpt_dict[training.MODEL_KEY])
+        self.logger.info(f"Model is initialized with precision {self.dtype}.")
 
         # Put model in eval mode.
         # Note: This will not disable the dropout applied in SDPA,
         # see https://github.com/pytorch/pytorch/issues/124464
         model.eval()
 
-        # Validate model was loaded in with the expected dtype.
-        training.validate_expected_param_dtype(
-            model.named_parameters(), dtype=self._dtype
-        )
-        logger.info(f"Model is initialized with precision {self._dtype}.")
-        return model
+        # Initialize tokenizer/transform
+        model_transform = config.instantiate(cfg.tokenizer)
 
-    @torch.no_grad()
-    def evaluate(self) -> None:
-        t1 = time.time()
-
-        model_eval_wrapper = _EvalWrapper(
-            self._model,
-            self._tokenizer,
-            device=self._device,
-            max_seq_length=self._max_seq_length,
-            batch_size=self._batch_size,
-            dtype=self._dtype,
+        # Finally, we setup the actual EvalWrapper class
+        eleuther_model_wrapper = (
```

Comment:
per our discussion earlier, this is simplified @felipemello1

Reply:
maybe worth logging?